### PR TITLE
fix: ACNA-2561 - app pack command should exclude hidden files

### PR DESCRIPTION
**Pull request by shazron**
Currently the files to be packed are defined by `npm pack --dry-run --json` and exclusions are defined by the `.gitignore` file. However, if the app folder does not have `.gitignore`, certain sensitive files may still be included. This PR will exclude these files no matter what.

These files are excluded from being packed:
1. any "hidden" file by convention (file that starts with a '.')
2. any junk file as defined by https://www.npmjs.com/package/junk

## How Has This Been Tested?

npm test

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Review by sandeep-paliwal — APPROVED** [review]

File: src/commands/app/pack.js [hunk 1]
```diff
@@ -250,6 +251,8 @@ class Pack extends BaseCommand {
     return files
       .map(file => file.path)
       .filter(file => !filesToExclude.includes(file))
+      .filter(junk.not) // no junk files like .DS_Store
+      .filter((file) => !/^\..*/.test(file)) // no files that start with a '.'
```
Comment (moritzraho) [inline src/commands/app/pack.js:255]: should we ignore gitignore files instead.. or is this a quick fix?

**Review by moritzraho — APPROVED** [review]

File: src/commands/app/pack.js (hunk 1 above)
Comment (moritzraho) [inline src/commands/app/pack.js:255]: Should we stdout/document this behavior somewhere?

Comment (shazron) [reply src/commands/app/pack.js:255]: good idea - maybe as a debug log. Maybe stdout for files we *include* would be better instead? (like when running npm pack in a npm package folder)

**Review by purplecabbage — APPROVED** [review]

File: src/commands/app/pack.js (hunk 1 above)
Comment (shazron) [reply src/commands/app/pack.js]: added in https://github.com/adobe/aio-cli-plugin-app/pull/734/commits/834b482413b787b1ae79860fe8012c9614e64996:
1. debug logging of omitted hidden dotfiles
2. debug logging of omitted junk files
3. stdout logging of files that will be packed

**Pull request merged**